### ISSUE TITLE
Xiangqi: it’s not possible to add engines BUG

### ISSUE THREAD

**Issue by lex312** [ml-research/liground#200]
I used Fairy-Stockfish.
Then I changed Fairy-Stockfish for a never version. -> Settings, clicked on the edit pencil, edit engine, new name, new select path. It works fine.

But I tried also: settings, clicked on the add button, add new engine, new name, new path, save.
Now when I click the button under engine settings to select another engine… nothing happens.
There is still only one engine available. The engine I have used before.

And when I try again to add the engine I tried before, then I get the answer that the name of the engine is already taken. But I still can’t see or use the engine.

**Comment by QueensGambit** [comment]
Are you sure that you downloaded the correct version of _Fairy-Stockfish_?
You need to download _fairy-stockfish-**largeboard**_x86-64-modern.exe_ or _fairy-stockfish-**largeboard**_x86-64-bmi2.exe_ for it to support Xiangqi.
Did you make sure, the engine runs from the command-line?

**Comment by lex312** [comment]
Yes that is fine.

My problem is that I can have only one engine in this gui.

Fairy Stockfish largeboard works fine. But if I want to use another Fairy Stockfish, then I need to delete the old one.

**Comment by QueensGambit** [comment]
If you set a different engine name, e.g.  _Fairy-Stockfish 13.1_, then you don't need to delete the old engine instance.

**Comment by lex312** [comment]
This doesn’t work.
I tried it with 6 different engines/compiles and of course with different names.

**Comment by TapioOfTapiola** [comment]
Right, an engine cannot be added if the Game Mode is Xiangqi. I noticed the engine will appear in the western chess section.
